### ISSUE TITLE
error "No report found to compare against."

### ISSUE THREAD

**Issue by 304NotModified** [codecov/support#173]
After the upgrade of CodeCov, we get in the PR's "No report found to compare against." . I saw it in multiple PR's

![image](https://cloud.githubusercontent.com/assets/5808377/14939143/7c29a970-0f3a-11e6-8692-a4a490d09ec3.png)

Is this a bug? or can we fix that?

e.g. https://github.com/NLog/NLog.Web/pull/54

**Comment by burabure** [comment]
this is also happening to us

**Comment by diwu1989** [comment]
I'm hitting this too, I found that for the last week, codecov has been basically unusable for us and actually causing a lot of annoyance for our team!

**Comment by stevepeak** [comment]
> PS below applies to PR's only. Commits on branches work slightly differently

This occurs when Codecov could not find a base commit to compare against and therefore does not know what to report.

```
• =  commit w/ report
. = commit w/out report

master  . • . •
pull         \ . . •
```

The base of the PR does not have reports and Codecov (CC) reports these statuses. We are planning a conceptual work around by allowing CC to search in history for the closest:

```
• =  commit w/ report
. = commit w/out report

chooen    v  (note how its -1 in history)
master  . • . •
pull         \ . . •
```

**Comment by 304NotModified** [comment]
What can we do about it?

**Comment by diwu1989** [comment]
can I force it to compare against `master` as a fallback?

**Comment by stevepeak** [comment]
@304NotModified well first make sure you have commits on the base branch of the pull request that are passing in CI (unless feature is disabled). Then if you do have commits on base branch: check to see if you are forking from commits that have reports in CC & passed in CI (which is generally a common/good practice). Then if all else contact us to review the project.

Optionally: You can disable the requirement of CI to pass.

There is good reason for the "complexity" (one may say). Simply put: Codecov does its best to reduce "positives". Lets say the PR base commit failed in CI and coverage was 50% as a result (but typically its 75%). A pull request starts and increase coverage to 60%.... no. Thats BS it actually reduced it by 15%.

**Comment by stevepeak** [comment]
@diwu1989 its a little more complicated then just is `X > Y`. Codecov does a diff on the code and analyses ever change to adjust coverage properly.

You can do this which basically skips ALL this talk above and just makes sure coverage hits a "target"

```
coverage:
  status:
    project: 
      default:
        target: 90%
```
> if you do this, then CC will never care about the "base" commit

**Comment by 304NotModified** [comment]
Well I looks like codecov gives wrong numbers/wrong errors.. If it stays like this, removal is better then wrong numbers. Previous version worked pretty good (with some small issues), but now it won't help anyone.

**Comment by stevepeak** [comment]
@304NotModified we tried to keep the same functionality from the last version. Which is true. If there are issues I'm happy to review.

From what I recall the only change was that we don't search to the nearest commit in the past if the PR base was missing... If that is the issue then I can look into adding it back.

**Comment by stevepeak** [comment]
@304NotModified can you give me some example owner/repo@sha of issues? I'll see if I can shed more light on this

**Comment by 304NotModified** [comment]
@stevepeak an example, 5 lines changed, but a lot of strange reports from codecov:

https://github.com/NLog/NLog/pull/1417

commit: https://github.com/NLog/NLog/pull/1417/commits/39f2851762806be47a51164aee2152602dfe3e60

results:
![image](https://cloud.githubusercontent.com/assets/5808377/14962273/4b109b7c-109e-11e6-99d5-53fc393f085f.png)

**Comment by stevepeak** [comment]
Hey, I do have an idea for a fix on this. I'll keep everyone posted on the progress. If all goes well I can release this w/in 24 hours.

**Comment by stevepeak** [comment]
@304NotModified what is the pull request # for this?

**Comment by 304NotModified** [comment]
1417?

**Comment by stevepeak** [comment]
Sorry didn't scroll up 👍

**Comment by stevepeak** [comment]
Check this out: https://codecov.io/gh/NLog/NLog/compare/e052b10...39f2851 tab to changes.

**Comment by 304NotModified** [comment]
@stevepeak that I guess the /patch? I should read the left column? (bit unclear)

But what about /changes and /project in https://github.com/codecov/support/issues/173#issuecomment-216307092?

This issue is really delaying our work.

**Comment by stevepeak** [comment]
@304NotModified For now you may want to disable the status, but I'm working hard on fixing this specific issues now. I'll be back with more details shortly.

**Comment by 304NotModified** [comment]
OK thanks for the feedback! 

We were very happy with codecov and we assume that will also the case in the future :)

**Comment by burabure** [comment]
hey @stevepeak, here's what I don't get: I send reports for every PR/branch that gets merged into our base branch, but it seems like codecov doesn't incorporate those reports as part of the base branch once they're merged.

What I had to do to actually have reports for commits in the base branch show up in codecov is upload them locally outside of CI. 

Am I supposed to send reports for each PR branch, then send them again for the main branch once the PR/branch is merged?

**Comment by stevepeak** [comment]
@304NotModified thank you so much for your patience 👍 honestly.

Planning on deploying this today.

**Comment by stevepeak** [comment]
Monitoring applied patch now. Thank you
